central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen_impl.go

```diff
@@ -99,11 +103,33 @@ type ResultEmail struct {
 func (rg *complianceReportGeneratorImpl) ProcessReportRequest(req *ComplianceReportRequest) {
 
 	log.Infof("Processing report request %s", req)
+
+	snapshot, found, err := rg.snapshotDS.GetSnapshot(req.Ctx, req.SnapshotID)
+	if err != nil {
+		log.Errorf("Unable to retrieve snapshot from the store: %v", err)
+		return
+	}
+	if !found {
+		log.Error("Unable to find snapshot in the store")
+		return
+	}
+
 	data := rg.getDataforReport(req)
 
 	zipData, err := format(data.ResultCSVs)
 	if err != nil {
 		log.Errorf("Error zipping compliance reports for scan config %s:%s", req.ScanConfigName, err)
+		snapshot.GetReportStatus().RunState = storage.ComplianceOperatorReportStatus_FAILURE
```

Comment:
codecov says this block isn't covered in a test.  Might be useful to cover this case since we are doing work here.

Reply:
I tried to increase the coverage of the report generation as much as I could without doing any major refactors. I've created [ROX-26740](https://issues.redhat.com/browse/ROX-26740) to address the missing coverage.